Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsBuildProvider.cs

```diff
@@ -9,65 +9,44 @@ namespace UnityEngine.InputSystem.Editor
 {
     internal class InputSettingsBuildProvider : IPreprocessBuildWithReport, IPostprocessBuildWithReport
     {
-        private InputSettings m_SettingsAddedToPreloadedAssets;
-
         public int callbackOrder => 0;
 
         public void OnPreprocessBuild(BuildReport report)
         {
             if (InputSystem.settings == null)
                 return;
 
-            var wasDirty = IsPlayerSettingsDirty();
-            m_SettingsAddedToPreloadedAssets = null;
+            // If there is no asset and we operate on temporary object,
+            // adding it would result in preloadedAssets containing null object "{fileID: 0}",
+            // which then we can't easily remove from the list in post build.
+            // Hence we ignore adding temporary objects to preloaded assets.
+            if (string.IsNullOrEmpty(AssetDatabase.GetAssetPath(InputSystem.settings)))
```

Comment:
Try using this instead https://docs.unity3d.com/ScriptReference/EditorUtility.IsPersistent.html